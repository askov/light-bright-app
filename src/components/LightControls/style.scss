.light-controls {
  margin-left: auto;
}
.control-button {
  cursor: pointer;
  border: none;
  padding: 10px;
  margin-right: 10px;
  transition-duration: 0.5s;
  text-transform: uppercase;
  background-color: #f2f2f2;
  color: #222;
  &:hover {
    background-color: #f44242;
    color: white;
  }
  &:last-child {
    margin-right: 0;
  }
}