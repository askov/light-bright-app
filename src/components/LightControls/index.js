import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

function LightControls(props) {
  return (
    <div className="light-controls">
      <button onClick={props.handleResetAll}
              className="control-button"
      >
        Reset all
      </button>
      <button onClick={props.handleReset}
              className="control-button"
      >
        Reset
      </button>
    </div>
  );
}

LightControls.propTypes = {
  handleResetAll: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
};

export default LightControls;
