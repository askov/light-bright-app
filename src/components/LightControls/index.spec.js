import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LightControls from './index';

describe('<LightControls />', () => {
  let props;
  let wrapper;

  const createWrapper = (props) => {
    return shallow(<LightControls {...props} />);
  }

  beforeEach(() => {
    props = {
      handleReset: jest.fn(),
      handleResetAll: jest.fn(),
    };
    wrapper = createWrapper(props);
  });

  test('renders correctly', () => {
    const tree = renderer
      .create(<LightControls {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders light controls container div', () => {
    expect(wrapper.find('div.light-controls').length).toBe(1);
  });

  // @todo: very specific (DOM) and looks wrong
  test('handles reset all button click', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(props.handleResetAll.mock.calls.length).toBe(1);
  });

  test('handles reset button click', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(props.handleReset.mock.calls.length).toBe(1);
  });

});