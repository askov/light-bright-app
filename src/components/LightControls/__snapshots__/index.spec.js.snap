// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LightControls /> renders correctly 1`] = `
<div
  className="light-controls"
>
  <button
    className="control-button"
    onClick={[MockFunction]}
  >
    Reset all
  </button>
  <button
    className="control-button"
    onClick={[MockFunction]}
  >
    Reset
  </button>
</div>
`;
